src/backend/distributed/planner/multi_router_planner.c

```diff
@@ -745,7 +745,17 @@ ExtractInsertPartitionValue(Query *query, Var *partitionColumn)
 												partitionColumn->varattno);
 	if (targetEntry != NULL)
 	{
-		Assert(IsA(targetEntry->expr, Const));
+		if (IsA(targetEntry->expr, Param))
+		{
+			ereport(ERROR, (errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
+							errmsg("cannot execute a prepared insert statement on "
```

Comment:
I think that's a pretty confusing message, and it's also not necessarily true. I wonder if we shouldn't just always error out if there's a param on the partition key instead. That'll be a lot easier for users to understand.